src/helpers/helpers.js

```diff
@@ -0,0 +1,14 @@
+const getPaymentStatus = (paymentDate) => {
+  const currentDate = new Date();
+  let dateLimit = new Date(paymentDate);
+  dateLimit.setMonth(dateLimit.getMonth() + 1);
+
+  if (paymentDate.getDate() !== dateLimit.getDate()) {
+    const dateCopy = new Date(paymentDate);
+    dateLimit = new Date(dateCopy.getFullYear(), dateCopy.getMonth() + 2, 1);
```

Comment:
Этот код не возвращает последний день следующего месяца
```suggestion
    dateLimit = new Date(dateCopy.getFullYear(), dateCopy.getMonth() + 2, -1);
```